x-pack/plugins/apm/server/routes/service_groups/lookup_services.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+import {
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { kqlQuery, rangeQuery } from '../../../../observability/server';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { Setup } from '../../lib/helpers/setup_request';
+import { MAX_NUMBER_OF_SERVICES_IN_GROUP } from '../../../common/service_groups';
+
+export async function lookupServices({
+  setup,
+  kuery,
+  start,
+  end,
+}: {
+  setup: Setup;
+  kuery: string;
+  start: number;
+  end: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search('lookup_services', {
+    apm: {
+      events: [
+        ProcessorEvent.metric,
+        ProcessorEvent.transaction,
+        ProcessorEvent.span,
+        ProcessorEvent.error,
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [...rangeQuery(start, end), ...kqlQuery(kuery)],
+        },
+      },
+      aggs: {
+        services: {
+          terms: {
+            field: SERVICE_NAME,
+            size: MAX_NUMBER_OF_SERVICES_IN_GROUP,
+          },
+          aggs: {
+            environments: {
```

Comment:
what's the point of the environments agg?

Reply:
This is to be able to list the environments in the kql filter service preview:
![Screen Shot 2022-03-09 at 9 35 28 AM](https://user-images.githubusercontent.com/1967266/157462948-93a6b640-0ffe-4155-a1bf-391dca70d115.png)